### PR TITLE
Integrate Orchestrator with NuGet.Services.Entities and NuGet.Services.Messaging.Email

### PR DESCRIPTION
**Pull request by scottbommarito**
https://github.com/nuget/engineering/issues/1812

This PR integrates the Orchestrator with the newest version of `NuGetGallery.Core`, which depends on `NuGet.Services.Entities`. It also integrates with the email sending job (through use of `NuGet.Services.Messaging.Email`'s `AsynchronousEmailMessageService`).

The vast majority of files changed here are related to configuring the `NuGet.Services.Entities` dependency. I also did a blind "remove all unused dependencies" in the entire solution, which is why you may see some unrelated files touched. I also fixed up some dependencies (we included some transitive dependencies, which caused restoring to warn).

Files of interest:

- `EmailConfiguration` now requires a `ServiceBusConfiguration` (for email sending job integration)
- `SmtpConfiguration` was removed--Orchestrator no longer uses SMTP
- `DiskMailSender` was removed--Orchestrator no longer needs an SMTP fallback
- `Validation.Orchestrator\Job` replaces the use of `CoreMarkdownMessageService` with `AsynchronousEmailMessageService` and sets up other dependencies needed by it
- `NuGet.Services.Validation.Orchestrator.csproj` now pulls in `NuGet.Services.Messaging.Email`
- `Validation.Common.Job.csproj` now pulls in latest `NuGetGallery.Core`

### CODE REVIEW

File: src/NuGet.Services.Revalidate/Initialization/PackageFinder.cs [hunk 1]
```diff
@@ -11,6 +11,7 @@
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
 using Newtonsoft.Json;
+using NuGet.Services.Entities;
 using NuGet.Versioning;
```
Comment (loic-sharma) [inline src/NuGet.Services.Revalidate/Initialization/PackageFinder.cs]: This looks unnecessary

File: src/Gallery.CredentialExpiration/Strings.Designer.cs [hunk 2]
```diff
@@ -8,10 +8,10 @@
 // </auto-generated>
 //------------------------------------------------------------------------------
 
-namespace Gallery.CredentialExpiration {
-    using System;
-    
-    
+namespace Gallery.CredentialExpiration
```
Comment (loic-sharma) [inline src/Gallery.CredentialExpiration/Strings.Designer.cs]: >CredentialExpiration [](start = 18, length = 20)

Considering this file is generated by a tool, I'd recommend undoing your changes here.

Comment (scottbommarito) [reply src/Gallery.CredentialExpiration/Strings.Designer.cs]: done both here and below

File: src/NuGet.Services.Validation.Orchestrator/Symbols/SymbolScanValidator.cs [hunk 3]
```diff
@@ -14,6 +14,7 @@
 using System;
 using System.Linq;
 using System.IO;
+using NuGet.Services.Entities;
```
[the remark this answers is not in the document]
Comment (scottbommarito) [reply src/NuGet.Services.Validation.Orchestrator/Symbols/SymbolScanValidator.cs]: fixed--looks like the ordering was broken before, and the tooling just made it worse haha

**Pull request merged**